src/payment/strategies/cba-mpgs/cba-mpgs-payment-strategy.ts

```diff
@@ -0,0 +1,199 @@
+    private _threeDSjs?: ThreeDSjs;
+    private _sessionId: string = '';
+
+    constructor(
+        store: CheckoutStore,
+        orderActionCreator: OrderActionCreator,
+        paymentActionCreator: PaymentActionCreator,
+        hostedFormFactory: HostedFormFactory,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _CBAMGPSScriptLoader: CBAMPGSScriptLoader,
+        private _locale: string
+    ) {
+        super(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            hostedFormFactory
+        );
+    }
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        await super.initialize(options);
+
+        const { methodId } = options;
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        const { clientToken, initializationData: { merchantId }, config: { is3dsEnabled, testMode } } = paymentMethod;
+
+        if (is3dsEnabled) {
+            this._threeDSjs = await this._CBAMGPSScriptLoader.load(testMode);
+
+            if (!this._threeDSjs) {
+                throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+            }
+
+            if (!clientToken || !merchantId ) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            this._sessionId = clientToken;
+
+            await this._threeDSjs.configure({
+                merchantId,
+                sessionId: this._sessionId,
+                callback: () => {
+                    if (this._threeDSjs?.isConfigured()) {
+                        return this._store.getState();
+                    }
+
+                    throw new PaymentMethodFailedError('Failed to configure 3DS API.');
+                },
+                configuration: {
+                    userLanguage: this._locale,
+                    wsVersion: 62,
+                },
+            });
+        }
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = payload;
+        const paymentData = payment && payment.paymentData;
+
+        if (!payment || !paymentData) {
+            throw new PaymentArgumentInvalidError(['payment.paymentData']);
+        }
+
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
+        const paymentMethod = getPaymentMethodOrThrow(payment.methodId);
+
+        const { is3dsEnabled } = paymentMethod.config;
+
+        if (is3dsEnabled) {
+            const newPaymentData = {
+                ...paymentData,
+                threeDSecure: { token: this._sessionId },
+            };
+
+            if (payload.payment) {
```

Comment:
How come we need to do this check again?  Since we check in line 85 that if payment exists in payload object?